reports/hosted-services/main.go

```diff
@@ -0,0 +1,199 @@
+		log.Fatalln(err.Error())
+	}
+
+	// Build the ingresses Map
+	ingressesMap := BuildIngressesMap(ingressList.Items)
+
+	// Add the ingress slice to the namespace Details map for all namespaces
+	for k, v := range nsDetailsMap {
+		if _, exist := ingressesMap[v.Name]; exist {
+			v.DomainNames = ingressesMap[v.Name]
+		}
+		nsDetailsMap[k] = v
+	}
+
+	jsonToPost, err := BuildJsonMap(nsDetailsMap)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	// Post json to hoowdaw api
+	err = hoodaw.PostToApi(jsonToPost, hoodawApiKey, &endPoint)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+}
+
+// GetNamespaces takes a Kubernetes clientset and returns all namespaces with type *v1beta1.IngressList and an error.
+
+func GetNamespaceDetails(ns v1.Namespace) namespace {
+
+	var namespaceDetails namespace
+
+	namespaceDetails.Name = ns.Name
+	namespaceDetails.Application = ns.Annotations["cloud-platform.justice.gov.uk/application"]
+	namespaceDetails.BusinessUnit = ns.Annotations["cloud-platform.justice.gov.uk/business-unit"]
+	namespaceDetails.DeploymentType = ns.Labels["cloud-platform.justice.gov.uk/environment-name"]
+	namespaceDetails.GithubURL = ns.Annotations["cloud-platform.justice.gov.uk/source-code"]
+	namespaceDetails.TeamName = ns.Annotations["cloud-platform.justice.gov.uk/team-name"]
+	namespaceDetails.TeamSlackChannel = ns.Annotations["cloud-platform.justice.gov.uk/slack-channel"]
+	namespaceDetails.DomainNames = []string{}
+	return namespaceDetails
+}
+
+// GetNamespaces takes a Kubernetes clientset and returns all namespaces with type *v1beta1.IngressList and an error.s
+func GetNamespaces(clientSet *kubernetes.Clientset) ([]v1.Namespace, error) {
+
+	namespaces, err := clientSet.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Fatalln("Can'list namespaces", err.Error())
+		return nil, err
+	}
+
+	return namespaces.Items, nil
+}
+
+// GetAllIngresses takes a Kubernetes clientset and returns all ingress with type *v1beta1.IngressList and an error.
+func GetAllIngresses(clientset *kubernetes.Clientset) (*v1beta1.IngressList, error) {
+	ingressList, err := clientset.NetworkingV1beta1().Ingresses("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return ingressList, nil
+}
+
+// BuildIngressesMap takes the Ingress list and return a map with key as namespace and value
+// with slices of string containing hosts urls
+func BuildIngressesMap(ingressItems []v1beta1.Ingress) map[string][]string {
+
+	ingressMap := make(map[string][]string, 0)
+
+	for _, i := range ingressItems {
+
+		for _, v := range i.Spec.TLS {
+			if len(v.Hosts) > 0 {
+				ingressMap[i.Namespace] = append(ingressMap[i.Namespace], v.Hosts[0])
+			}
+		}
+	}
+	return ingressMap
+}
+
+// BuildJsonMap takes a map with namespce key and namespace struct as value, sort the map, flatten to a
+// slice and return a json encoded map
+func BuildJsonMap(hostedservices map[string]namespace) ([]byte, error) {
```

Comment:
Is it worth putting this in a helper package. I imagine other packages finding this very useful.

Reply:
We have a pkg directory here:
https://github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/tree/main/reports/pkg/hoodaw

Reply:
The issue is building this info: https://github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/blob/hostedservices-go/reports/hosted-services/main.go#L190
For this report the type of value is []map[string]namespace, but for others it could be different []map[string]<any type>